modules/account-lib/src/coin/dot/transferBuilder.ts

```diff
@@ -1,26 +1,195 @@
+      }
+    } else if (decodedTxn.method?.name === MethodNames.Proxy) {
+      const txMethod = decodedTxn.method.args as unknown as ProxyArgs;
+      const real = txMethod.real;
+      const forceProxyType = txMethod.forceProxyType;
+      const decodedCall = utils.decodeCallMethod(rawTransaction, {
+        registry: this._registry,
+        metadataRpc: this._metadataRpc,
+      });
+      const amount = `${decodedCall.value}`;
+      const to = decodedCall.dest.id;
+      const validationResult = ProxyTransactionSchema.validate({ real, forceProxyType, amount, to });
+      if (validationResult.error) {
+        throw new InvalidTransactionError(`Proxy Transaction validation failed: ${validationResult.error.message}`);
+      }
+    }
   }
 
   /** @inheritdoc */
-  protected fromImplementation(rawTransaction: any): Transaction {
-    throw new NotImplementedError('method not implemented');
+  protected fromImplementation(rawTransaction: string): Transaction {
+    const tx = super.fromImplementation(rawTransaction);
+    if (this._method?.name === MethodNames.TransferKeepAlive) {
+      const txMethod = this._method.args as TransferArgs;
+      this.amount(txMethod.value);
+      this.to({ address: txMethod.dest.id });
+    } else if (this._method?.name === MethodNames.Proxy) {
+      const txMethod = this._method.args as ProxyArgs;
+      this.real({ address: txMethod.real });
+      this.forceProxyType(txMethod.forceProxyType);
+      const decodedCall = utils.decodeCallMethod(rawTransaction, {
+        registry: this._registry,
+        metadataRpc: this._metadataRpc,
+      });
+      if (!decodedCall.value || !decodedCall.dest) {
+        throw new InvalidTransactionError(
+          `Invalid Proxy Transaction Method: ${this._method?.name}. Expected transferKeepAlive`,
+        );
+      }
+      this.amount(`${decodedCall.value}`);
+      this.to({ address: decodedCall.dest.id });
+    } else {
+      throw new InvalidTransactionError(
+        `Invalid Transaction Type: ${this._method?.name}. Expected a transferKeepAlive or a proxy transferKeepAlive transaction`,
+      );
+    }
+    return tx;
   }
 
   /** @inheritdoc */
-  protected signImplementation(key: BaseKey): Transaction {
-    throw new NotImplementedError('method not implemented');
+  validateTransaction(_: Transaction): void {
+    super.validateTransaction(_);
+    this.validateFields(this._to, this._amount, this._real, this._forceProxyType);
+  }
+
+  private validateFields(to: string, amount: string, real?: string, forceProxyType?: string): void {
+    const validationResult = forceProxyType
+      ? ProxyTransactionSchema.validate({ to, amount, real, forceProxyType })
+      : TransferTransactionSchema.validate({ amount, to });
+
+    if (validationResult.error) {
+      throw new InvalidTransactionError(`Proxy Transaction validation failed: ${validationResult.error.message}`);
```

Comment:
Keep clear here that now you are calling validate fields inside validate transactions so, the error message on validate fields support now transferKeepAlive and proxy transaction, and error message only mention proxy.

Reply:
good catch, will fix